lib/spack/spack/audit.py

```diff
@@ -694,25 +694,21 @@ def _unknown_variants_in_directives(pkgs, error_cls):
                 )
 
         # Check "depends_on" directive
-        for _, triggers in pkg_cls.dependencies.items():
-            triggers = list(triggers)
-            for trigger in list(triggers):
-                vrn = spack.spec.Spec(trigger)
-                errors.extend(
-                    _analyze_variants_in_directive(
-                        pkg_cls, vrn, directive="depends_on", error_cls=error_cls
-                    )
+        for trigger in pkg_cls.dependencies:
+            vrn = spack.spec.Spec(trigger)
+            errors.extend(
+                _analyze_variants_in_directive(
+                    pkg_cls, vrn, directive="depends_on", error_cls=error_cls
                 )
+            )
 
-        # Check "patch" directive
-        for _, triggers in pkg_cls.provided.items():
-            triggers = [spack.spec.Spec(x) for x in triggers]
-            for vrn in triggers:
-                errors.extend(
-                    _analyze_variants_in_directive(
-                        pkg_cls, vrn, directive="patch", error_cls=error_cls
-                    )
+        # Check "provides" directive
+        for when_spec in pkg_cls.provided.keys():
```

Comment:
Should be the default:
```suggestion
        for when_spec in pkg_cls.provided:
```

Reply:
done